Sources/SQLKit/SQLDatabase.swift

```diff
@@ -1,14 +1,94 @@
+///
+/// - Note: Most of ``SQLDatabase``'s functionality is relatively low-level. Clients of SQLKit
+///   who want to query a database should use the higher-level API rooted at ``SQLQueryBuilder``.
+///
+/// Example of manually constructing and executing a query from expressions without a query builder:
+///
+/// ```swift
+/// var select = SQLSelect()
+/// select.columns = [SQLColumn(SQLIdentifier("x"))]
+/// select.tables = [SQLIdentifier("y")]
+/// select.predicate = SQLBinaryExpression(
+///     left: SQLColumn(SQLIdentifier("z")),
+///     op: SQLBinaryOperator.equal,
+///     right: SQLLiteral.numeric("1")
+/// )
+/// var resultRows: [SQLRow] = []
+/// let sqlDb = // obtain an SQLDatabase from somewhere
+///
+/// try await sqlDb.execute(sql: select, resultRows.append(_:))
+/// // Executed query: SELECT x FROM y WHERE z = 1, as represented in the database's SQL dialect.
+/// ```
+///
+/// It should almost never be necessary for a client to call ``SQLDatabase/execute(sql:_:)-90wi9``
+/// directly; such a need usually indicates a design flaw or functionality gap in SQLKit itself.
 public protocol SQLDatabase {
+    /// The ``Logger`` to be used for logging all SQLKit operations relating to a given database.
     var logger: Logger { get }
+    
+    /// The ``NIOCore/EventLoop`` used for asynchronous operations on a given database. If there is no
+    /// specific ``NIOCore/EventLoop`` which handles the database (such as because it is a connection
+    /// pool which assigns loops to connections at point of use, or because the underlying implementation
+    /// is based on Swift Concurrency or some other asynchronous execution technology), it is recommended
+    /// to return an event loop from ``NIOCore/EventLoopGroup/any()``.
     var eventLoop: EventLoop { get }
+    
+    /// The version number the connection reports for itself, provided as a type conforming to the
+    /// ``SQLDatabaseReportedVersion`` protocol. If the version number is not applicable (such as for
+    /// a connection pool dispatch wrapper) or not yet known, `nil` may be returned. Version numbers
+    /// may also change at runtime (for example, if a connection is auto-reconnected after a remote
+    /// update), or even become unknown again after being known.
+    ///
+    /// - Warning: This version number has nothing to do with ``SQLKit`` or (usually) of the driver
+    ///   implementation for the database, nor does it represent any data stored within the database;
+    ///   it is the version of the database implementation _itself_ (such as of a MySQL server or
+    ///   `libsqlite3` library). A significant part of the motivation to finally add this property comes
+    ///   from a larger desire to enable customizing a given ``SQLDialect``'s configuration based on the
+    ///   actual feature set available at runtime instead of having to hardcode a "safe" baseline.
+    var version: SQLDatabaseReportedVersion? { get }
```

Comment:
Is it worth renaming this to `databaseVersion` or similar to avoid any ambiguity

Reply:
I'm already unhappy about the repetition of "database" in the name of the type, honestly 😕

Reply:
Fair - I'm just thinking of potential future versions (no pun intended) where we expose other versions, but I'm happy with this as is